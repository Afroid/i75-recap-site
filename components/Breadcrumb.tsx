import Link from "next/link";
import { TestIds } from "@/lib/testIds";

interface BreadcrumbProps {
  year: string;
  week: string;
}

/**
 * A simple breadcrumb navigation component:
 * - Root link to home
 * - Year link to recap list
 * - Current week label (non-clickable)
 */
export default function Breadcrumb({ year, week }: BreadcrumbProps) {
  const crumbs = [
    { href: "/", label: "I75 League" },      // Home
    { href: `/recaps/${year}`, label: year },   // Year list
    { href: "", label: week },                // Current page
  ];

  return (
    <nav
      data-testid={TestIds.BREADCRUMB_NAV}
      aria-label="Breadcrumb"
      className="text-sm text-gray-600 mb-6"
    >
      <ol data-testid={TestIds.BREADCRUMB_LIST} className="flex flex-wrap">
        {crumbs.map((crumb, idx) => (
          <li
            key={idx}
            data-testid={`${TestIds.BREADCRUMB_ITEM}-${idx}`}
            className="flex items-center"
          >
            {crumb.href ? (
              // Clickable link for all but last crumb
              <Link
                data-testid={`${TestIds.BREADRCRUMB_LINK}-${idx}`}
                href={crumb.href}
                className="hover:underline text-blue-600"
              >
                {crumb.label}
              </Link>
            ) : (
              // Current page label
              <span
                data-testid={TestIds.BREADCRUMB_CURRENT}
                className="text-gray-900 font-semibold"
              >
                {crumb.label}
              </span>
            )}
            {idx !== crumbs.length - 1 && (
              // Separator line after I75 League and before Week X (Last element in breadcrumb)
              <span
                data-testid={`${TestIds.BREADCRUMB_SEPARATOR}-${idx}`}
                className="px-2 text-gray-400"
              >
                /
              </span>
            )}
          </li>
        ))}
      </ol>
    </nav>
  );
}
