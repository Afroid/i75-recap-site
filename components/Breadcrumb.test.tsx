import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Breadcrumb from "@/components/Breadcrumb";

// Stub next/link so it just renders a plain <a>
jest.mock("next/link", () => {
  return {
    __esModule: true,
    default: ({ href, children }: { href: string; children: React.ReactNode }) => (
      <a href={href}>{children}</a>
    ),
  };
});

describe("Breadcrumb", () => {
  const year = "2023";
  const week = "5";

  it("renders the correct links and current crumb", () => {
    render(<Breadcrumb year={year} week={week} />);

    // First crumb: home link
    const homeLink = screen.getByRole("link", { name: "I75 League" });
    expect(homeLink).toHaveAttribute("href", "/");

    // Second crumb: year link
    const yearLink = screen.getByRole("link", { name: year });
    expect(yearLink).toHaveAttribute("href", `/recaps/${year}`);

    // Final crumb: plain text span (not clickable)
    const weekSpan = screen.getByText(week);
    expect(weekSpan.tagName).toBe("SPAN");
    expect(weekSpan).toHaveClass("font-semibold");
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<Breadcrumb year={year} week={week} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
