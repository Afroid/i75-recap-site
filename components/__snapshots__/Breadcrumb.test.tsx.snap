// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb matches the snapshot 1`] = `
<DocumentFragment>
  <nav
    aria-label="Breadcrumb"
    class="text-sm text-gray-600 mb-6"
    data-testid="breadcrumb-nav"
  >
    <ol
      class="flex flex-wrap"
      data-testid="breadcrumb-list"
    >
      <li
        class="flex items-center"
        data-testid="breadcrumb-item-0"
      >
        <a
          href="/"
        >
          I75 League
        </a>
        <span
          class="px-2 text-gray-400"
          data-testid="breadcrumb-separator-0"
        >
          /
        </span>
      </li>
      <li
        class="flex items-center"
        data-testid="breadcrumb-item-1"
      >
        <a
          href="/recaps/2023"
        >
          2023
        </a>
        <span
          class="px-2 text-gray-400"
          data-testid="breadcrumb-separator-1"
        >
          /
        </span>
      </li>
      <li
        class="flex items-center"
        data-testid="breadcrumb-item-2"
      >
        <span
          class="text-gray-900 font-semibold"
          data-testid="breadcrumb-current"
        >
          5
        </span>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;
